uplink/config.go

```diff
@@ -4,13 +4,21 @@
 package uplink
 
 import (
+	"io/ioutil"
+	"strings"
 	"time"
 
+	"github.com/zeebo/errs"
+	"gopkg.in/spacemonkeygo/monkit.v2"
+
 	"storj.io/storj/internal/memory"
+	libuplink "storj.io/storj/lib/uplink"
```

Comment:
Package uplink probably shouldn't depend on libuplink. `uplink` is the lower-level with all the bits and pieces for custom streaming/metainfo/encryption etc. making it depend on `libuplink` seems weird.

Reply:
Right now all `uplink` has is a config struct and some helpers to create some objects from it.

I don't mind moving the helper somewhere else, though, but something has to create a Scope from an `uplink.Config`.

Reply:
Some of the config part shouldn't be part of it as well, it's conflating the uplink and the needs of the binary.

I suspect we are missing a uplinkflags package under `cmd/uplink/`, or `cmd/internal`, for these things.

Reply:
Sure, I'd love to see a refactoring of the `uplink` package from someone who has a vision of what it's fore lol

Reply:
Ok, let's leave as is, once we finish metainfo refactor, I'll clean it up. Otherwise there are too many refactorings in-flight at the same time.